packages/nextra-theme-docs/src/navbar.tsx

```diff
@@ -11,12 +13,88 @@ import { useConfig } from './config'
 import Search from './search'
 import Flexsearch from './flexsearch'
 import { GitHubIcon, DiscordIcon, XIcon, MenuIcon } from 'nextra/icons'
-import { Item, PageItem } from './utils/normalize-pages'
+import { Item, PageItem, MenuItem } from './utils/normalize-pages'
 
 interface NavBarProps {
   isRTL?: boolean | null
   flatDirectories: Item[]
-  items: PageItem[]
+  items: (PageItem | MenuItem)[]
+}
+
+const NavbarMenuLink = React.forwardRef((props: any, ref) => {
```

Comment:
I guess it's redundant to create this simple component used in 1 place

Reply:
Oh it's because I have to use forwardRef to make the menu automatically hide itself after clicking the link: https://headlessui.com/react/menu#integrating-with-next-js